snapcraft/application.py

```diff
@@ -141,6 +141,14 @@ def _configure_services(self, platform: str | None, build_for: str | None) -> No
         self.services.set_kwargs("package", platform=platform, build_for=build_for)
         super()._configure_services(platform, build_for)
 
+        self.services.set_kwargs(
+            "lifecycle",
+            cache_dir=self.cache_dir,
+            work_dir=self._work_dir,
+            build_plan=self._build_plan,
+            project_path=self._resolve_project_path(None),
```

Comment:
I don't think this is a workable solution; trying to resolve the project path here breaks cases that don't have projects at all, like:
![image](https://github.com/canonical/snapcraft/assets/1951299/b2a3a79e-2d88-4cf1-99b9-6db5491bf8fc)

Reply:
Fixed to account for non-existing projects, failing tests are again in the expected set.